tpe/plugin/src/SimulationFeatures.cc

```diff
@@ -58,6 +62,51 @@ void SimulationFeatures::WorldForwardStep(
     }
   }
   world->Step();
+  this->WriteRequiredData(_h);
+  this->Write(_h.Get<JointPositions>());
+}
+
+void SimulationFeatures::Write(JointPositions &/*_positions*/) const
+{
+  // TODO(adlarkin) implement this? Is this needed?
+}
+
+void SimulationFeatures::Write(WorldPoses &_poses) const
+{
+  // cache link poses from the previous iteration
+  // (using a static variable for now instead of a member variable
+  // since this method has to be const)
+  static std::unordered_map<std::size_t, math::Pose3d> prevLinkPoses;
```

Comment:
Same comment from the DART implementation applies here about using a static variable

Reply:
Resolved in https://github.com/ignitionrobotics/ign-physics/pull/223/commits/baeb7e6606c21ade93ced6af5d96c6139f41d878